source/Mlos.Python/mlos/Optimizers/OptimumDefinition.py

```diff
@@ -45,6 +45,11 @@ class OptimumDefinition(Enum):
 
     BEST_SPECULATIVE_WITHIN_CONTEXT = 'best_speculative_within_context'
 
+    # This mode works to select the points that given a function f(series) is able to output a utility value for that series. This will often be related to
+    # time series approximation however it shouldn't be limited to that.
+    #
+    BEST_SERIES_ERROR_FUNCTION_MINIMIZATION = 'best_series_error_function_minimization'

```

Comment:
In the optimization problem the user specifies that they want to match time series. And the time-series utility functions compute the relevant goodness of fit metrics and their statistics, and the optimizer will try to spit out either the: best observation (series that most closely matched the target at least once), best predicted value, etc... I don't see why we need a new optimum type for this.